cas-server-core-audit/src/main/java/org/jasig/cas/audit/spi/TicketOrCredentialPrincipalResolver.java

```diff
@@ -115,4 +119,15 @@ private String resolveArgument(final Object arg1) {
         LOGGER.debug("Unable to determine the audit argument. Returning [{}]", UNKNOWN_USER);
         return UNKNOWN_USER;
     }
+
+    /**
+     * Default implementation that simply returns principal#id.
+     */
+    public static class DefaultPrincipalIdProvider implements PrincipalIdProvider {
```

Comment:
Perfect. In that case, may I suggest then that we move this class into its own file? I realize it's small enough to be included as an inner class, but it might be better if we gave it its own personality.